rule/utils_test.go

```diff
@@ -30,3 +31,122 @@ func TestIsCallToExitFunction(t *testing.T) {
 		})
 	}
 }
+
+func TestHasUpperCaseFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"Exit", true},
+		{"fmt", false},
+		{"_SOME_PRIVATE_CONST_2", true},
+		{"", false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := hasUpperCase(tt.varName); got != tt.expected {
+				t.Errorf("hasCaps(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func BenchmarkHasUpperCaseRE(b *testing.B) {
+	var anyUpperCaseRE = regexp.MustCompile(`[A-Z]`)
+	for i := 0; i <= b.N; i++ {
+		input := "HeLlo_WoRlD"
+		_ = anyUpperCaseRE.MatchString(input)
+	}
+}
+
+func BenchmarkHasUpperCase(b *testing.B) {
+	for i := 0; i <= b.N; i++ {
+		input := "HeLlo_WoRlD"
+		_ = hasUpperCase(input)
+	}
+}
+
+func TestHasLowerFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"Exit", true},
+		{"fmt", true},
+		{"_SOME_PRIVATE_CONST_2", false},
+		{"HELLO_WORLD123", false},
+		{"Hello_World", true},
+		{"", false},
+		{"INVALID-CHAR", false},
+		{"abc123", true},
+		{"123", false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.varName, func(t *testing.T) {
+			if got := hasLower(tt.varName); got != tt.expected {
+				t.Errorf("hasLower(%s) = %v; want %v", tt.varName, got, tt.expected)
+			}
+		})
+	}
+}
+
+func BenchmarkAllCapsRE(b *testing.B) {
+	var allUpperCaseRE = regexp.MustCompile(`^_?[A-Z][A-Z\d]*(_[A-Z\d]+)*$`)
+	for i := 0; i <= b.N; i++ {
+		input := "_SOME_PRIVATE_CONST_2"
+		_ = allUpperCaseRE.MatchString(input)
+	}
+}
+
+func BenchmarkAllCaps(b *testing.B) {
+	for i := 0; i <= b.N; i++ {
+		input := "_SOME_PRIVATE_CONST_2"
+		_ = hasUpperCase(input)
+	}
+}
+
+func TestIsUpperConstFunction(t *testing.T) {
+	tests := []struct {
+		varName  string
+		expected bool
+	}{
+		{"SOME_CONST_2", true},
+		{"__FOO", false},
+		{"__", false},
+		{"X509B", true},
+		{"FOO", true},
+		{"1FOO", false},
+		{"_FOO123_BAR456", true},
+		{"A1_B2_C3", true},
+		{"A1_b2", false},
+		{"FOO_", false},
```

Comment:
About this one @chavacava @alexandear don't you think we could consider it's an uppercase constants ?

Reply:
@ccoVeille did you think that ```A1_b2``` or ```FOO_``` should be processed as a proper consts?

Reply:
I don't know. But I think it's worth raising the point.

If we had to make a detector about the fact something is in exact screamimg snake case, I would say these are invalid.

But then, here we are dealing with code that are not respecting Go standard.

Here it's a linter rule, I would report them


Also think about these technology acronym

- 3DES
- 386
- 2FA

Why wouldn't we report the following constants as uppercase constants

- 3DES_CIPHER
- 386_ARCH
- 2FA_TTL

While the following are reported

- CIPHER_3DES
- ARCH_386
- TTL_2FA

Please note these could be part of a separate issue and PR.

I'm raising these points because I think about it while reviewing the PR.